Celeste.Mod.mm/Patches/Spikes.cs

```diff
@@ -9,11 +11,21 @@
 
 namespace Celeste {
     public class patch_Spikes : Spikes {
-        public patch_Spikes(Vector2 position, int size, Directions direction, string type) 
+        public patch_Spikes(Vector2 position, int size, Directions direction, string type)
             : base(position, size, direction, type) {
             // no-op. MonoMod ignores this - we only need this to make the compiler shut up.
         }
 
+        public extern void orig_ctor(EntityData data, Vector2 offset, Directions dir);
+
+        [MonoModConstructor]
+        public void ctor(EntityData data, Vector2 offset, Directions dir) {
+            orig_ctor(data, offset, dir);
+            if (data.Has("attachToSolid") && !data.Bool("attachToSolid")) {
```

Comment:
This causes two dictionary lookups, you can make use of the 2nd parameter to Bool instead.
Imo it would also be nice to inject this condition directly into the ctor, both for speed and also to not introduce another orig_ method, but its not the end of the world